packages/jellyfish-api-core/__tests__/category/loan/paybackLoan.test.ts

```diff
@@ -763,3 +763,192 @@ describe('paybackLoan failed', () => {
+  const testing = Testing.create(container)
+
+  const dusdLoanAmount = 5000
+  const tslaLoanAmount = 10
+  const loanSchemeId = 'scheme'
+  let vaultId: string
+  let tslaVaultId: string
+  let vaultOwnerAddress: string
+  let dusdInterestAmountBefore: BigNumber
+  let dusdInterestPerBlock: BigNumber
+  let tslaTakeLoanBlockHeight: number
+
+  async function setupForDUSDLoan (): Promise<void> {
+    vaultOwnerAddress = await testing.generateAddress()
+    await testing.token.dfi({ amount: 1000000, address: vaultOwnerAddress })
+    await testing.generate(1)
+
+    // setup oracle
+    const oracleAddress = await testing.generateAddress()
+    const priceFeeds = [
+      { token: 'DFI', currency: 'USD' },
+      { token: 'TSLA', currency: 'USD' },
+      { token: 'DUSD', currency: 'USD' }
+    ]
+
+    const oracleId = await testing.rpc.oracle.appointOracle(oracleAddress, priceFeeds, { weightage: 1 })
+    await testing.generate(1)
+    const timestamp = Math.floor(new Date().getTime() / 1000)
+    await testing.rpc.oracle.setOracleData(oracleId, timestamp, { prices: [{ tokenAmount: '1@DFI', currency: 'USD' }] })
+    await testing.rpc.oracle.setOracleData(oracleId, timestamp, { prices: [{ tokenAmount: '1000@TSLA', currency: 'USD' }] })
+    await testing.rpc.oracle.setOracleData(oracleId, timestamp, { prices: [{ tokenAmount: '1@DUSD', currency: 'USD' }] })
+    await testing.generate(1)
+
+    // setup collateral token
+    await testing.rpc.loan.setCollateralToken({
+      token: 'DFI',
+      factor: new BigNumber(1),
+      fixedIntervalPriceId: 'DFI/USD'
+    })
+    await testing.generate(1)
+
+    // setup loan token
+    await testing.rpc.loan.setLoanToken({
+      symbol: 'DUSD',
+      fixedIntervalPriceId: 'DUSD/USD'
+    })
+    await testing.generate(1)
+
+    await testing.rpc.loan.setLoanToken({
+      symbol: 'TSLA',
+      fixedIntervalPriceId: 'TSLA/USD'
+    })
+
+    // setup loan scheme
+    await testing.rpc.loan.createLoanScheme({
+      minColRatio: 150,
+      interestRate: new BigNumber(3),
+      id: loanSchemeId
+    })
+    await testing.generate(1)
+
+    // create vault
+    vaultId = await testing.rpc.loan.createVault({
+      ownerAddress: vaultOwnerAddress,
+      loanSchemeId: loanSchemeId
+    })
+
+    await testing.generate(1)
+    await testing.container.waitForPriceValid('DFI/USD')
+
+    // deposite collateral
+    await testing.rpc.loan.depositToVault({
+      vaultId: vaultId,
+      from: vaultOwnerAddress,
+      amount: '100000@DFI'
+    })
+    await testing.generate(1)
+
+    // take DUSD as loan
+    const dusdTakeLoanBlockHeight = await testing.rpc.blockchain.getBlockCount()
+    await testing.rpc.loan.takeLoan({
+      vaultId: vaultId,
+      amounts: `${dusdLoanAmount}@DUSD`,
+      to: vaultOwnerAddress
+    })
+    await testing.generate(1)
+
+    dusdInterestPerBlock = new BigNumber(netInterest * dusdLoanAmount / (365 * blocksPerDay)).decimalPlaces(8, BigNumber.ROUND_CEIL)
```

Comment:
```suggestion
    dusdInterestPerBlock = new BigNumber(netInterest * dusdLoanAmount / (365 * blocksPerDay))
```
never ceil the `InterestPerBlock`, you will need it with full precision for further calculations.